Implement new forcing in marrmot class

SarahAlidoost opened the pull request:
closes #92  #91

SarahAlidoost committed:
implement new forcing in marrmot class

SarahAlidoost committed:
remove os.pathlike and unused imports

SarahAlidoost committed:
replace isoformat with strftime

SarahAlidoost committed:
fix date_string in tests

SarahAlidoost committed:
add load_foreign and re-run the marrmotm01 example notebook

SarahAlidoost committed:
revise docstring of forcing in init function

SarahAlidoost committed:
Merge branch 'master' into update_forcing_in_marrmotm01_class

SarahAlidoost committed:
refactor tests regarding sample_marrmot_forcing_file

SarahAlidoost committed:
add parent to file path in sample_marrmot_forcing_file

sverhoeven commented on examples/MarrmotM01.ipynb:
```diff
@@ -34,7 +34,7 @@
   {
```
<p>Typo in heading 3</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/eWaterCycle/ewatercycle/pull/95/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

sverhoeven commented on ewatercycle/models/marrmot.py:
```diff
@@ -261,11 +254,8 @@ def parameters(self) -> Iterable[Tuple[str, Any]]:
             ('solver', self.solver),
-            ('start time', self.forcing_start_time.isoformat()),
-            ('end time', self.forcing_end_time.isoformat()),
+            ('start time', self.forcing_start_time.strftime("%Y-%m-%dT%H:%M:%SZ")),
```
Looks a bit weird to have a format string, however to get UTC timezone this is needed. Also `self.forcing_*_time` gets sets using get_time which forces UTC so OK to hard-code timezone here.

sverhoeven commented on tests/models/test_marrmotm01.py:
```diff
@@ -25,8 +26,19 @@ def forcing_file(self, sample_marrmot_forcing_file):
+                                end_time='1992-12-31T00:00:00Z',
+                                forcing_info={
+                                    'forcing_file': str(Path(forcing_file).name)
+                                })
```
Indent is one space to many
```suggestion
                               directory=str(Path(forcing_file).parent),
                               start_time='1989-01-01T00:00:00Z',
                               end_time='1992-12-31T00:00:00Z',
                               forcing_info={
                                   'forcing_file': str(Path(forcing_file).name)
                               })
```

sverhoeven reviewed (approved):
After commenting out Lisflood in models/__init__py the tests ran OK.

Notebook also ran OK.

Some small typos, style points, but overall nice work.

SarahAlidoost committed:
revise indent

Co-authored-by: Stefan Verhoeven <s.verhoeven@esciencecenter.nl>

SarahAlidoost committed:
fix typo in heading 3 in the notebook

SarahAlidoost commented:
> After commenting out Lisflood in models/__init__py the tests ran OK.
> 
> Notebook also ran OK.
> 
> Some small typos, style points, but overall nice work.

Many thanks for reviewing, I addressed your comments.  Shall I merge this PR?
